task/prepare.go

```diff
@@ -114,8 +115,10 @@ func getPlaybackProfiles(assetVideoSpec *livepeerAPI.AssetVideoSpec) ([]livepeer
 		return nil, fmt.Errorf("no video track found in asset spec")
 	}
 	filtered := make([]livepeerAPI.Profile, 0, len(allProfiles))
-	for _, profile := range allProfiles {
-		if profile.Height <= video.Height && profile.Bitrate < int(video.Bitrate) {
+	for _, baseProfile := range allProfiles {
+		profile := effectiveProfile(baseProfile, video)
+		if profile.Height <= video.Height && profile.Bitrate < int(video.Bitrate) &&
+			profile.Height > minVideoDimensionPixels && profile.Width > minVideoDimensionPixels {
```

Comment:
WDYT of actually breaking up these checks just so it's more readable? Can either create temporary variables giving names to the conditions or have multiple `if` checks with `continue` statements here. e.g. WDYT?
```suggestion
        lowerQualityThanSrc := profile.Height <= video.Height && profile.Bitrate < int(video.Bitrate)
        compliesToMinSize := profile.Height >= minVideoDimensionPixels && profile.Width >= minVideoDimensionPixels
		if lowerQualityThanSrc && compliesToMinSize {
```